distributed/diagnostics/graph_layout.py

```diff
@@ -45,6 +51,12 @@ def __init__(self, scheduler):
     def update_graph(
         self, scheduler, dependencies=None, priority=None, tasks=None, **kwargs
     ):
+        if len(self.scheduler.tasks) > self.max_items:
+            logger.debug(
+                "Ignoring submitted graph for dashboard visualization since limit was reached."
+            )
+            return
```

Comment:
With this patch it will simply ignore every submitted graph once the limit is reached. That means we'd never be able to visualise the ignored graphs. If this is acceptable, I don't see a new to reset any state. If this behaviour is not acceptable, I'll need to search for another solution.

Reply:
Right, but if we then release a bunch of those tasks and fall under this limit again then we continue on as before.  I think that this is the policy that we want, but I think that today it will fail if some of those tasks from when we paused are still in the scheduler.  New tasks might refer to them, but they won't be represented in the Layout object's state.